app/models/collection.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
```

Comment:
This "class" feels still really hacky to me.. open to suggestions about how to better structure this.